### DISCUSSION TITLE
[Proposal] Prisma.io integration

### DISCUSSION THREAD

**Discussion by marceloverdijk** [withastro/roadmap#245] [Proposal]
It was mentioned on Discord as well, but it would be great to have Astro integration with Prisma.io.
Something like `astro add prisma`

I created this proposal/discussion so we can refer to this and track progress if it is decided to create such an integration.

**Comment by unbiased-dev** [comment]
Just a random passerby with a question...but, what would a prisma integration do more than the current setup where you just install prisma and use it inside the astro component script bit? What part of the manual integration is currently a pain point?

**Reply by marceloverdijk** [reply]
Actually a good point .
I'm quite new to Astro and found some people talking on Discord on such an integration but couldn't find anything on GitHub.
Maybe there is nothing to provide from Astro , hence the discussion here.

**Comment by Kellytom** [comment]
Prisma Accelerate Example: Astro Starter
This project showcases how to use Prisma ORM with Prisma Accelerate in a Astro application. It [demonstrates](safari-reader://github.com/prisma/prisma-examples/blob/latest/accelerate/astro-starter/src/pages/route.ts#L13-16) every available [caching strategy in Accelerate](https://www.prisma.io/docs/accelerate/caching#cache-strategies).

To successfully run the project, you will need the following:

The connection string of a publicly accessible database.
Your Accelerate connection string (containing your Accelerate API key) which you can get by enabling Accelerate in a project in your [Prisma Data Platform](https://pris.ly/pdp) account (learn more in the [docs](https://www.prisma.io/docs/platform/about#api-keys)).
Clone the repository, navigate into it and install dependencies:

git clone git@github.com:prisma/prisma-examples.git --depth=1
cd prisma-examples/accelerate/astro-starter
npm install
2. Configure environment variables

Copy the .env.example env file in the root of the project directory:

cp .env.example .env
Now, open the .env file and set the DIRECT_URL and DATABASE_URL environment variables with the values of your connection string and your Accelerate connection string respectively... See GitHub repo
